ui/perfherder/alerts/AlertTable.jsx

```diff
@@ -119,6 +123,28 @@ export default class AlertTable extends React.Component {
     this.setState({ filteredAlerts });
   };
 
+  updateAssignee = async newAssigneeUsername => {
+    const { updateAlertSummary, updateViewState } = this.props;
+    const { alertSummary } = this.state;
+
+    const { data, failureStatus } = await updateAlertSummary(alertSummary.id, {
+      assignee_username: newAssigneeUsername,
+    });
+
+    if (!failureStatus) {
+      alertSummary.assignee_username = newAssigneeUsername;
+      this.setState({ alertSummary });
+    } else {
+      updateViewState({
+        errorMessages: [
+          `Failed to set new assignee "${newAssigneeUsername}". (${data})`,
+        ],
+      });
+    }
+
+    return { failureStatus };
```

Comment:
Why put `failureStatus` in an object?

Reply:
I'm curious why this function is in `AlertTable` and passed as a prop to `AlertHeader` and then `Assignee` when it's only used in `Assignee`?

Reply:
> Why put `failureStatus` in an object?

`Assignee` needs to know if the update was successful or not, so it can update it's UI accordingly *(toggle edit mode off)*.

Reply:
> I'm curious why this function is in `AlertTable` and passed as a prop to `AlertHeader` and then `Assignee` when it's only used in `Assignee`?

It's not really `Assignee`'s responsibility to do the nitty-gritty stuff of updating the assignee of a summary. `Assignee`'s main purpose is to nicely display the assignee name on top of an `AlertTable` & let `AlertTable` know that an update has been triggered.
`AlertTable` is actually much more concerned about the current assignee & whether it updates or not.

Reply:
> > Why put `failureStatus` in an object?
> 
> `Assignee` needs to know if the update was successful or not, so it can update it's UI accordingly _(toggle edit mode off)_.

Right, I understand that part. What I'm asking is why return it in an object instead of doing `return failureStatus`?